### DISCUSSION TITLE
Server.ts -> Server.js

### DISCUSSION THREAD

**Discussion by omarkhatibgg** [remix-run/remix#1651] [Q&A]
Hi Everyone,

I wanted to start with Remix by migrating one of my custom react server written using Express.js and koa.js.

The problem is that everything is written using typescript and it seems server.js is out of the scope of Remix right now. 

did anyone was able to do an example about using typescript with Remix on the `server.js` file?

and how this did work with the development environment and file watching? 

if not, did anyone was able to append something to the context in the loader ? or able to use middlewares?

**Comment by danprince** [comment]
Also keen on migrating an existing CRA + Express codebase to Remix and this is the part of the process that is the least clear.

I need to share code between some of the non-React server components (e.g. websocket handlers) but currently it looks like Remix stays hands off this code, meaning that I would need to bring my own compiler and bundler.

Would love some more clarity around this in the [TypeScript guide](https://remix.run/docs/en/v1/guides/typescript).

Looks like there's some existing discussion in https://github.com/remix-run/remix/issues/81 too.

**Reply by omarkhatibgg** [reply]
It seems that the server.js part is out of the scope right now, I just found this in Express Starter https://github.com/remix-run/starter-express/issues/18

In the mean time, I think the best solution to check what did Kent do with his website here: https://github.com/kentcdodds/kentcdodds.com

**Comment by cliffordfajardo** [comment]
I have not tried you're use case but would adding the following line at the top of your Typescript file help?
```
// @ts-nocheck
```

Reference:
- https://stackoverflow.com/questions/55632954/ignore-all-errors-in-a-typescript-file

**Reply by omarkhatibgg** [reply]
I don't want to ignore the errors, it's more about the ability to right my server in Typescript and able to import my already written code without having to rewrite it in Js or bringing my own compiler
